resilience4j2/src/main/java/com/linecorp/armeria/resilience4j/circuitbreaker/client/Resilience4JCircuitBreakerClientHandler.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.resilience4j.circuitbreaker.client;
+
+import static java.util.Objects.requireNonNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerClientHandler;
+import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerMapping;
+import com.linecorp.armeria.client.circuitbreaker.ClientCircuitBreakerGenerator;
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.circuitbreaker.CircuitBreakerCallback;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+
+/**
+ * A {@link CircuitBreakerClientHandler} implementation for use with Resilience4j's {@link CircuitBreaker}.
+ *
+ * <pre>{@code
+ * // for HttpRequest
+ * CircuitBreakerRule rule = CircuitBreakerRule.onException();
+ * CircuitBreakerClientHandler<HttpRequest> handler = Resilience4JCircuitBreakerClientHandler.of(
+ *     Resilience4jCircuitBreakerMapping.builder()
+ *                                      .perHost()
+ *                                      .registry(CircuitBreakerRegistry.custom()
+ *                                                                      ...
+ *                                                                      .build())
+ *                                      .build());
+ * WebClient.builder()
+ *          .decorator(CircuitBreakerClient.newDecorator(handler, rule))
+ *          ...
+ *
+ * // for RpcRequest
+ * CircuitBreakerRuleWithContent rule = ...;
+ * CircuitBreakerClientHandler<RpcRequest> handler = Resilience4JCircuitBreakerClientHandler.of(
```

Comment:
I'm not sure if the `T` type parameter for `Request` makes the API type-safe. It looks a bit verbose to use. `ClientCircuitBreakerGenerator` already takes `Request` as a parameter. How about removing the `T` type parameter from `CircuitBreakerClientHandler` and inline `Request` to the method argument?
```java
CircuitBreakerCallback tryRequest(ClientRequestContext ctx, Request req);
```

Reply:
I'm guessing if we do this we can't use the following pattern 😅 I've updated the javadocs though 🙏 

```
final Function<? super HttpClient, CircuitBreakerClient> circuitBreakerDecorator =
        CircuitBreakerClient.newDecorator( // expects HttpRequest as the parameterized type, but is Request
                Resilience4JCircuitBreakerClientHandler.of(mapping), rule);
```

Reply:
Couldn't we change `CircuitBreakerClientHandler<HttpRequest>` to `CircuitBreakerClientHandler`? It is still an unstable API.
https://github.com/line/armeria/blob/e495f0d9bd4ea744c2e85249067ac78678005ae6/core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerClient.java#L116

Reply:
I think you're right 👍   Good idea as always 🙇